docs/pages/components/ops.md

```diff
@@ -10,7 +15,358 @@ Defined in header: `<pressio/ops.hpp>`
+  ModeB /* modeB */,
+  const scalar_type alpha,
+  const A_type& A,
+  const B_type& B,
+  const scalar_type beta,
+  C_type& C)
+```
+- Multiplies matrices `A` and `B` computing @f$C := \alpha A B + \beta C@f$ if both modes are `pressio::nontranspose`. Use `pressio::transpose` to transpose input matrices of choice.
+@endparblock
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template <class scalar_type, class A_type, class C_type>
+void product(
+  ::pressio::transpose /* modeA */,
+  ::pressio::nontranspose /* modeB */,
+  const scalar_type alpha,
+  const A_type& A,
+  const scalar_type beta,
+  C_type& C)
+```
+- Computes @f$C := \alpha A^T A + \beta C@f$
+@endparblock
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template <class C_type, class scalar_type, class A_type>
+C_type product(
+  ::pressio::transpose /* modeA */,
+  ::pressio::nontranspose /* modeB */,
+  const scalar_type alpha,
+  const A_type& A)
+```
+- Computes and returns @f$C = \alpha A^T A@f$
+@endparblock
+
+
+## Utilities
+
+#### extent()
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template<class T, class IndexType>
+Traits<T>::size_type extent(const T& objectIn, const IndexType i)
+```
+- Returns the number of elements that `objectIn` has in `i`-th dimension (zero based).
+@endparblock
+
+#### matching_extents::compare()
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template<typename T1, typename T2>
+bool matching_extents<T1, T2>::compare(const T1& a, const T2& b)
+```
+- Returns `true` if objects `a` and `b` have same number of elements in each dimension.
+@endparblock
+
+#### clone()
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template <class T>
+T clone(const T& src)
+```
+- Creates a copy of `src` object.
+@endparblock
+
+#### deep_copy()
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template<typename T1, typename T2>
+void deep_copy(T1& dest, const T2& src)
+```
+- Copies `src` to `dest` assuring a deep copy.
+@endparblock
+
+#### fill() and set_zero()
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template <typename T>
+void fill(T& o, Traits<T>::scalar_type value)
+
+template <typename T>
+void set_zero(T& o)
+```
+- `fill()` populates `o` (matrix or vector) with given scalar `value`.
+- `set_zero()` populates `o` (matrix or vector) with zeros.
+@endparblock
+
+@m_class{m-block m-default}
+
+#### resize()
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template <typename T>
+void resize(T& o, const Traits<T>::size_type newSize)
+```
+- Resizes vector `o` into `newSize` number of elements.
+@endparblock
+
+@m_class{m-block m-default}
+
+@parblock
+```cpp
+template <typename T>
+void resize(
+  T& o,
+  const Traits<T>::size_type newRows,
+  const Traits<T>::size_type newCols)
+```
+- Resizes matrix `o` into `newCols` x `newRows` number of elements.
+@endparblock
+
+## Type Traits
+
+These pseudo-ops provide simple and convenient form for selected [type traits](md_pages_components_type_traits.html):
```

Comment:
can you please remove the stuff below ? I don't want to expose this, it is just for internal use. 
I thought it was all inside `impl`

Reply:
Section removed!